src/core/layout/qgslayoutitemlegend.h

```diff
@@ -49,6 +50,72 @@ class CORE_EXPORT QgsLegendModel : public QgsLayerTreeModel
     QVariant data( const QModelIndex &index, int role ) const override;
 
     Qt::ItemFlags flags( const QModelIndex &index ) const override;
+
+    /**
+     * Set the pointer to the layoutlegend expressioncontext
+     * \param econtext pointer to the context
+     * \since QGIS 3.8
+     */
+    void setLayoutExpContext( QgsExpressionContext *econtext );
+
+    /**
+     * Is used to control QgsLayoutItemLegend::refresh, otherwise segfault may occur when updating featurecount hashes (symbol expressions).
+     * \since QGIS 3.8
+     */
+    bool refreshable = true;
+
+  signals:
+
+    /**
+     * Emitted to refresh the legend once counting is done.
+     * \since QGIS 3.8
+     */
+    void refreshLegend();
```

Comment:
Would it work to emit a `dataChanged()` signal instead of introducing a new signal here?

Reply:
Possibly.

Reply:
but in the c++ code this signal is used exclusively to trigger the refresh function that already exists, and I'm not sure if the datachanged or the refresh function is optimal as the refresh also take into account the atlas(?)

Reply:
I think having a dedicated signal might be better than a datachanged.